library/init/meta/quote.lean

```diff
@@ -9,53 +9,51 @@ import init.meta.tactic
 open tactic
 
 meta class has_quote (α : Type) :=
-(quote : α → expr)
+(quote : α → pexpr)
 
-@[inline] meta def quote {α : Type} [has_quote α] : α → expr :=
+@[inline] meta def quote {α : Type} [has_quote α] : α → pexpr :=
 has_quote.quote
 
-private meta def c (n : name) := expr.const n []
-
-meta instance : has_quote nat := ⟨expr.mk_prenum_macro⟩
+meta instance : has_quote nat := ⟨pexpr.of_raw_expr ∘ expr.mk_prenum_macro⟩
 
 meta instance : has_quote char :=
-⟨λ ⟨n, pr⟩, expr.app (expr.const ``char.of_nat []) $ quote n⟩
+⟨λ ⟨n, pr⟩, ``(char.of_nat %%(quote n))⟩
 
 @[priority std.priority.default + 1]
-meta instance : has_quote string := ⟨expr.mk_string_macro⟩
+meta instance : has_quote string := ⟨pexpr.of_raw_expr ∘ expr.mk_string_macro⟩
```

Comment:
We can provide `pexpr.mk_string : string -> pexpr`, and delete `expr.mk_string_macro`.

We should avoid `pexpr.of_raw_expr`. This is a temporary hack.